File: Maths/ChkPrime.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    int t = 2;
    while (t--)
    {
        int n;
        int count = 0;
        vector<int> ls;
        cin >> n;
        for (int i = 1; i <= sqrt(n); i++)
        {
            if (n % i == 0)
            {
                count++;
                if (n / i != i)
                {
                    count++;
                }
            }
        }if(count==2){
            cout << "YES" << endl;
        }else{
            cout << "NO" << endl;
        }
    }
}

File: Maths/ChkPrime.py
import math

# for _ in range(2):
#     n = int(input("Enter the number: "))
for n in range(1,10):
    count=0
    for i in range(1,int(math.isqrt(n))+1):
        if n % i == 0:
            count += 1
            if n // i != i:
                count += 1
    if count == 2:
        print(f"Yes {n}")
    else:
        print("No") 